io/types.d.ts

```diff
@@ -1,57 +1,203 @@
+  /** Specifies whether to enable or disable the "auto-teardown" feature
+   * (see section below).
+   * @default true
    */
   teardown?: boolean;
-  /** 
+  /**
    * Specifies whether to persist this socket so it can be reused
-   * (see [vuexModule](https://nuxt-socket-io.netlify.app/vuexModule)). 
+   * (see [vuexModule](https://nuxt-socket-io.netlify.app/vuexModule)).
    * @default false
    */
   persist?: boolean;
   /** Specifies the property in [this] component that will be used
-   *  to contain the socket status (referring to an object). 
-   * @default 'socketStatus' 
+   *  to contain the socket status (referring to an object).
+   * @default 'socketStatus'
    */
   statusProp?: string;
-  /** Specifies the timeout in milliseconds for an emit event, 
-   * after which waiting for the emit response will be canceled. 
-   * @default undefined 
+  /** Specifies the timeout in milliseconds for an emit event,
+   * after which waiting for the emit response will be canceled.
+   * @default undefined
    */
   emitTimeout?: number;
-  /** 
-   * Specifies the property in [this] component that will be used 
-   * to contain emit errors (see section below). 
+  /**
+   * Specifies the property in [this] component that will be used
+   * to contain emit errors (see section below).
    * (referring to this.emitErrors, an object of arrays)
-   * @default 'emitErrors' 
+   * @default 'emitErrors'
    */
   emitErrorsProp?: string;
   namespaceCfg?: {
     emitters?: Array<string | object>;
     listeners?: Array<string | object>;
   };
+  /**
+   * @default 'ioApi'
+   */
+  ioApiProp?: string;
+  /**
+   * @default 'ioData'
+   */
+  ioDataProp?: string;
+  /**
+   * @default []
+   */
+  apiIgnoreEvts?: Array<string>;
+  serverAPI?: {
+    label: string;
+    version: number;
+    evts?:  Record<string, any>;
+    methods?:  Record<string, any>;
+  };
+  clientAPI?: {
+    label: string;
+    version: number;
+    evts?: Record<string, any>;
+    methods?: Record<string, any>;
+  };
+  vuex?: NuxtSocketVueOptions;
 }
 
-// export interface NuxtSocket extends SocketIOClient.Socket {}
+interface NuxtSocketConfig {
+  /**
+   * Recommended for all sockets, but required for any non-default socket.
+   */
+  name?: string;
+  /**
+   * URL for the Socket.IO server.
+   * @default window.location
+   */
+  url?: string;
+  /**
+   * Determines which socket is used as the default when creating new sockets
+   * with `nuxtSocket()`
+   * @default true // for the first entry in the array
+   */
+  default?: boolean;
+  /**
+   * Options to let you sync incoming events to a Vuex store and emit events
+   * on Vuex store changes. These options will override settings from your Nuxt
+   * configuration.
+   */
+  vuex?: NuxtSocketVueOptions;
+  /**
+   * Socket.IO namespaces configuration. Supports an arrow syntax in each entry
+   * to help describe the flow (with pre/post hook designation support too).
+   */
+  namespaces?: Record<string, {
```

Comment:
If using a namespaceCfg type as mentionned above, this can be Record<string, namespaceCfg> or similar.